pylint_nautobot/q_search_filter.py

```diff
@@ -0,0 +1,48 @@
+"""Check for use of SearchFilter on NautobotFilterSets instead of custom `q` search functions."""
+
+from astroid import Assign, AssignName, ClassDef
+from pylint.checkers import BaseChecker
+
+from .utils import is_version_compatible
+
+_META_CLASSES = {
+    "nautobot.extras.filters.NautobotFilterSet": ">1",
+    "nautobot.apps.filters.NautobotFilterSet": ">1",
+}
+
+
+class NautobotUseSearchFilterChecker(BaseChecker):
+    """Visit NautobotFilterSet subclasses and check for use of `q = SearchFilter`, instead of `q = django_filters.CharField`."""
+
+    version_specifier = ">=2,<3"
```

Comment:
FWIW `SearchFilter` was implemented in v1.3, if I'm reading the release-notes correctly.

Reply:
Ken stated this check only needed to work for 2+.

Reply:
agreed.